### ISSUE TITLE
The VertxTestContext.failing() method should release the latch 

### ISSUE THREAD

**Issue by cvgaviao** [vert-x3/vertx-junit5#77]
### Version

4.0.0-SNAPSHOT

### Context
I have created a test in order to ensure that my server is failing to startup due to a missing configuration.
So the MainVerticle is being failed with an exception. Then the MainVerticleIT is supposed to pass ok since it is waiting for a failure, but the test is wrongly failing with a timeout instead.

### Do you have a reproducer?

https://github.com/cvgaviao/vertx-junit5-investigation

### Steps to reproduce

1. just run the included test

### Extra
I was able to pass the test using the failing method with parameter...
```
vertx.deployVerticle(new MainVerticle(), testContext.failing((o) -> {
            testContext.completeNow();
        }));
```

**Comment by cvgaviao** [comment]
@jponge, I added a reproducer as you required. ;)

**Comment by julianladisch** [comment]
Thank you for the reproducer and the extra.

This test

    @Test
    void foo(Vertx vertx, VertxTestContext testContext) {
      vertx.deployVerticle(new MainVerticle(), testContext.failing());
    }

times out with this timeout message:

> The test execution timed out. Make sure your asynchronous code includes calls to either VertxTestContext#completeNow(), VertxTestContext#failNow() or Checkpoint#flag()

This works [as documented](https://vertx.io/docs/vertx-junit5/java/#_use_any_assertion_library):

> Warning: Unlike completing(), calling succeeding and failing methods can only make a test fail (e.g., succeeding gets a failed asynchronous result). To make a test pass you still need to call completeNow, or use checkpoints

If we have two verticles and both should fail because of missing configuration we use this test:

    @Test
    void foo(Vertx vertx, VertxTestContext testContext) {
      vertx.deployVerticle(new MainVerticle(), testContext.failing(exception1 -> {
        vertx.deployVerticle(new SecondVerticle(), testContext.failing(exception2 -> {
          testContext.completeNow();
        }));
      }));
    }

If the first `failing` were automatically completing the test then SecondVerticle wouldn't be tested. Therefore `failing` works correctly and as documented, this issue should be closed.

**Comment by cvgaviao** [comment]
@julianladisch, many thanks for the explanation. I agree with it ;)
But note that in your example, you are using the failing method that has parameters, not the one without parameters which is the one I'm using... 
I was not able yet to see the point of having a single `failing()` that does not complete itself... 
could you show me an example of it?

**Comment by julianladisch** [comment]
@cvgaviao Thank you for your report, this has been fixed by #80.
